skiko/src/commonMain/kotlin/org/jetbrains/skiko/GraphicsApi.kt

```diff
@@ -29,3 +29,17 @@ enum class GpuPriority(val value: String) {
         fun parseOrNull(value: String): GpuPriority? = GpuPriority.values().find { it.value == value }
     }
 }
+
+enum class FrameBuffering {
+    DEFAULT,
+    DOUBLE,
+    TRIPLE
```

Comment:
> refactor with [nextDrawable](https://developer.apple.com/documentation/quartzcore/cametallayer/1478172-nextdrawable?language=objc) being called on IO dispatcher and non-blocking introspection*

@elijah-semyonov, do we really need a separate thread?

As far as I understand Triple buffering, there a 3 buffers and we can swap them without blocking and moving operations to another thread:
```
updating buffer <-> back buffer <-> front buffer
```

Reply:
If we don't want to sleep on `nextDrawable` - yes, we need extra threads, it's blocking.

Reply:
Discussed online - there can be a way to not wait for `nextDrawable` in a case of triple buffering, if we manually control acquired/completed `drawables`.

I.e. we can postpone commiting the third buffer, if 2 previous one are still not completed. And reuse it, when there will a urgent request for redrawing (`redrawImmediately` on resize)

Reply:
I'm not sure that I fully understand the idea. I would be happy to discuss the ideas on how to tackle this problem offline, or on call. So don't hesitate to approach me, I'm in office in every weekday

Reply:
> I would be happy to discuss the ideas

If you are working on improving resize, this should be considered. As it doesn't block, and might not have an issue of missing frame (a theory).